### PR TITLE
Add Upgrades key tests.

### PR DESCRIPTION
**Pull request by samuel-williams-da**
- [x] Daml3-script query tests
  - This tests both that daml3-script upgrades/downgrades the key/payloads it receives, and that the ledger-api can give back contracts of a different version when filtering by key. 
- [x] ExerciseByKey command
  - Ensures both daml3-script submission logic handles exerciseByKey, and that canton (CommandPreprocessing) can exercise by an upgraded/downgraded key (it can't)
- [x] Fetching by key
- [x] ExerciseByKey in Update.

Found that using the `exerciseByKey` command, with or without package ids included, gives a CONTRACT_NOT_FOUND error when using a key of a different version. Interestingly, the error includes a global key, and not a shared key, which is likely the reason for this behaviour.

Also adds "unsafe" variants of the vet and unvet dar methods, which don't wait, so we can wait for the transactions to land in bulk

### CODE REVIEW

File: daml-script/test/daml/upgrades/UpgradesTest.daml
Comment (samuel-williams-da) [inline daml-script/test/daml/upgrades/UpgradesTest.daml:78]: Improved cleanup using unsafe variant, so we only delay 2s, rather than 4s previously for `broken` tests

File: daml-script/test/daml/upgrades/UpgradesTest.daml
Comment (simonmaxen-da) [inline daml-script/test/daml/upgrades/UpgradesTest.daml:660]: Is this used ?

File: daml-script/test/daml/upgrades/UpgradesTest.daml
Comment (simonmaxen-da) [inline daml-script/test/daml/upgrades/UpgradesTest.daml:627]: Is this used

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ScriptF.scala
Comment (simonmaxen-da) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ScriptF.scala:445]: We plan to drop this flag. Upgradability will need to be determined by language version. (fyi)

**Review by simonmaxen-da — APPROVED** [review]
lgtm

File: daml-script/test/daml/upgrades/UpgradesTest.daml
Comment (samuel-williams-da) [reply daml-script/test/daml/upgrades/UpgradesTest.daml:660]: Ah, forgot to remove these, my bad

**Comment by samuel-williams-da** [comment]
These tests aren't running until the PackageName logic in canton is merged to canton3, and following that, #18214 is merged.
#18236 will be a good indicator of whether this suite will pass once the logic is transferred, so we can wait on that to merge this.

**Pull request merged**
